### ISSUE TITLE
Link to mdn documentation if the dart api documentation is empty

### ISSUE THREAD

**Issue by kasperpeulen** [dart-lang/dart-pad#255]
I think we should provide a link to the mdn documentation if the dart dom api documentation is empty. I propose the following procedure:

* If the element has dart documentation, don't show a mdn link.
* Else, look for the DomName metadata, and check if the page https://developer.mozilla.org/en-US/docs/Web/API/${@domname} doesn't give errors
  * If the page doesn't give errors, use that link
  * Else, use the link: https://developer.mozilla.org/en/search?q=${@domname}

We should probably tell if the link is pointing to a search page or to the specific page with information. I think that in many cases the domname will point to the correct mdn information, but there are cases where this doesn't work. This seem to be about differences between the chrome IDL and firefox.

For example. If you have the code:

```
ButtonElement button = new ButtonElement;
button.style<cursor>
```

Then the domname metadata will point to:
https://developer.mozilla.org/en-US/docs/Web/API/Element.style
but this page doesn't exist, it should point to:
https://developer.mozilla.org/en-US/docs/Web/API/HtmlElement.style

Ideally, we could make a mapping for where the chrome IDL and firefox seems to have differences.  Thoughts @devoncarew @lukechurch ?

**Comment by sethladd** [comment]
This is probably a feature of the doc generation for dart:html, not a feature for the dartdoc tool itself. The dartdoc tool shouldn't have custom logic for dart:html or MDN.

I like the idea, but it should live somewhere else. For example, what processing step can we introduce in the generation of dart:html ?

**Comment by sethladd** [comment]
oops, commented in wrong context. :)

**Comment by lukechurch** [comment]
@devoncarew Any suggest wrt this?

**Comment by kasperpeulen** [comment]
Wow, it seems the in the meantime, Firefox has created redirect for Element.style. this can't be coincidence right?

**Comment by devoncarew** [comment]
Haven't had time to look a the associated PR, but My thinking is that this should be solved in the dart:html library itself. __It__ should have great documentation; either Dart specific, or linking to MDN if that's the right thing to do for an element.

I'm not sure solving this at the dartpad level is the right thing.

**Comment by kasperpeulen** [comment]
@devoncarew I do agree that this should be solved in the dart:html library (ideally). But I'm afraid that it may quite a long time before this is solved at the dart:html side, and this is really a feature I would like to have, as a dartpad user myself. 

I haven't ever done serious javascript, and so my dom knowledge is not that great. As the dart dom api is lacking much documentation, in 90% of the cases you have to search for external dom documentation. In that case, it is of course great to know hat the native dom name is. So I think we should give at least that information.

For classes, api.dartlang.org does that as well btw. I don't think it is unreasonable to show the complete native dom name also for class members at dartlang, as the doc snippits we provide are a bit more isolated.

But then, if you are going to give the native dom name, and if the user is going to use that native dom name, probably the user is going to search for external documentation. What my script basicly does, is already checking for the user, if it can find some external resources (if there is no dart documentation).

First it checks if https://developer.mozilla.org/en-US/docs/Web/API/$domName exists. Where domName for example could be HTMLTextAreaElement.autofocus. If this page doesn't exists, it tries to extract the className of the domName, in this case, HTMLTextAreaElement, and then it checks if 
https://developer.mozilla.org/en-US/docs/Web/API/$className exists.

And last if both pages doesn't exists, is just shows the message, search for [HTMLTextAreaElement.autofocus] at MDN. 

I think this can do very little harm, while the benefit can be really great for people. Of course, if dart:html will improve so much that this isn't needed, then we can just remove this script right ?

**Comment by devoncarew** [comment]
fixed by #257; thanks!
